nntrainer/src/neuralnet.cpp

```diff
@@ -448,12 +448,16 @@ int NeuralNetwork::train_run() {
 
     int count = 0;
 
+    TensorDim inputDim = getInputDimension();
+    TensorDim labelDim{batch_size, 1, 1,
+                       layers.back()->getOutputDimension().width()};
+    std::shared_ptr<float> in, label;
```

Comment:
how about creating a tensor here itself with the given shape. Then passing float shared_ptr from `tensor.getData()` to `getDataFromBuffer()`?

Reply:
good idea. It'll be done :)

Reply:
Changed to use `float * out, float * label` that is already allocated from a tensor.